src/parser/Parse_Table.hh

```diff
@@ -107,60 +101,53 @@ struct Keyword {
+ * when the input token is not a keyword, END, EXIT, or ERROR, the parse table
+ * reports an error, then attempts recovery by reading additional tokens until
+ * it encounters either a keyword it recognizes or an END, EXIT, or ERROR.
+ * During this recovery, no additional errors are reported, to avoid swamping
+ * the user with additional messages that are unlikely to be helpful.  If
+ * recovery is successful (a keyword is recognized in the token stream) parsing
  * resumes as normal.
  *
- * User parse routines may also encounter errors.  These may be reported
- * to the Token_Stream through either the Report_Syntax_Error or
- * Report_Semantic_Error functions.  The former is used when there is
- * an error in the input syntax, as when a keyword is encountered when
- * a numerical value is expected.  In this case, an exception is thrown
- * that is caught by the Parse_Table, which then attempts error recovery
- * as described above.  The Report_Semantic_Error indicates syntactically
- * correct input that is nonetheless unacceptable, as when a numerical
- * value appears where it is expected but violates some constraint, such
- * as positivity.  The error is reported but input processing is not
- * interrupted.
+ * User parse routines may also encounter errors.  These may be reported to the
+ * Token_Stream through either the Report_Syntax_Error or Report_Semantic_Error
+ * functions.  The former is used when there is an error in the input syntax, as
+ * when a keyword is encountered when a numerical value is expected.  In this
+ * case, an exception is thrown that is caught by the Parse_Table, which then
+ * attempts error recovery as described above.  The Report_Semantic_Error
+ * indicates syntactically correct input that is nonetheless unacceptable, as
+ * when a numerical value appears where it is expected but violates some
+ * constraint, such as positivity.  The error is reported but input processing
+ * is not interrupted.
  *
- * For parsers designed to receive input from the console, the default
- * behavior just described is not entirely appropriate. Console parsers are
- * supported by treating any OTHER token whose text character is a semicolon
- * as an empty keyword.  By default, semicolons are treated as whitespace and
- * the parser will never see such a token.  A console stream can choose to
- * convert an endline or other terminator to a semicolon token to force
- * processing.
+ * For parsers designed to receive input from the console, the default behavior
+ * just described is not entirely appropriate. Console parsers are supported by
+ * treating any OTHER token whose text character is a semicolon as an empty
+ * keyword.  By default, semicolons are treated as whitespace and the parser
+ * will never see such a token.  A console stream can choose to convert an
+ * endline or other terminator to a semicolon token to force processing.
  */
-
-class DLL_PUBLIC_parser Parse_Table
```

Comment:
Why are we getting rid of DLL_PUBLIC_parser? No longer needed?

Reply:
(There are a few places where this is either removed or added)

Reply:
`DLL_PUBLIC_*` macros are no longer needed for classes and functions because recent versions of CMake automatically export those symbols for us.  

CMake doesn't do this for global data types (static class data or global scope data).  If such data is meant to be accessible via the shared object library, then these variables must still be marked as `DLL_PUBLIC_*`.